google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -79,61 +139,148 @@ public File defaultDockerfile(DeploymentArtifactType deploymentArtifactType) {
-          + "type for automatic deployment");
-    }
-    return new AppEngineDeployAction(
+      DeploymentOperationCallback callback) {
+    AppEngineDeploy deploy = new AppEngineDeploy(
         this,
         loggingHandler,
-        project,
-        artifactToDeploy,
         deploymentConfiguration,
-        wrapCallbackForUsageTracking(deploymentCallback, ConfigType.AUTO, artifactType)
-    );
+        wrapCallbackForUsageTracking(callback, deploymentConfiguration, artifactToDeploy));
+
+    if (environment == Environment.APP_ENGINE_STANDARD) {
+      AppEngineStandardStage standardStage = new AppEngineStandardStage(
+          this,
+          loggingHandler,
+          artifactToDeploy);
+
+      return new AppEngineStandardDeployRunner(deploy, standardStage);
+    } else {
+      AppEngineFlexibleStage flexibleStage = new AppEngineFlexibleStage(
+          this,
+          loggingHandler,
+          artifactToDeploy,
+          deploymentConfiguration);
+
+      return new AppEngineFlexibleDeployRunner(deploy, flexibleStage);
+    }
   }
 
-  @NotNull
   @Override
-  public AppEngineStopAction createStopAction(
+  public File createStagingDirectory(LoggingHandler loggingHandler) throws IOException {
+    File stagingDirectory = FileUtil.createTempDirectory(
+        "gae-mvm" /* prefix */,
+        null /* suffix */,
+        true  /* deleteOnExit */);
+    loggingHandler.print(
+        "Created temporary staging directory: " + stagingDirectory.getAbsolutePath() + "\n");
+
+    return stagingDirectory;
+  }
+
+  @Override
+  public CloudSdk createSdk(
       LoggingHandler loggingHandler,
-      AppEngineDeploymentConfiguration deploymentConfiguration,
-      String moduleToStop,
-      String versionToStop,
-      UndeploymentTaskCallback undeploymentTaskCallback) {
-    return new AppEngineStopAction(
-        this,
-        loggingHandler,
-        deploymentConfiguration,
-        moduleToStop,
-        versionToStop,
-        undeploymentTaskCallback
+      ProcessStartListener startListener,
+      ProcessOutputLineListener stdErrListener,
+      ProcessOutputLineListener stdOutListener,
+      ProcessExitListener exitListener) {
+    if (credentialsPath == null) {
+      loggingHandler.print(GctBundle.message("appengine.action.credential.not.found") + "\n");
+      throw new AppEngineException("Failed to create application default credentials.");
+    }
+
+    CloudToolsPluginInfoService pluginInfoService =
+        ServiceManager.getService(CloudToolsPluginInfoService.class);
+
+    return new CloudSdk.Builder()
+        .sdkPath(getGcloudCommandPath())
+        .async(true)
+        .addStdErrLineListener(stdErrListener)
+        .addStdOutLineListener(stdOutListener)
+        .exitListener(exitListener)
+        .startListener(startListener)
+        .appCommandCredentialFile(credentialsPath)
+        .appCommandMetricsEnvironment("gcloud-intellij")
```

Comment:
I'd expect the gcloud-intellij value should come from pluginInfoService as well.